external/fv3fit/fv3fit/sklearn/_random_forest.py

```diff
@@ -128,84 +125,8 @@ def dump(self, path: str) -> None:
     def load(cls, path: str) -> "RandomForest":
         return RandomForest.from_sklearn_wrapper(SklearnWrapper.load(path))
 
-
-class _RegressorEnsemble:
-    """
-    Ensemble of RandomForestRegressor objects that are each trained on a separate
-    batch of data.
-    """
-
-    def __init__(
-        self,
-        base_regressor,
-        n_jobs,
-        regressors: Sequence[sklearn.base.BaseEstimator] = None,
-    ) -> None:
-        self.base_regressor = base_regressor
-        self.regressors = regressors or []
-        self.n_jobs = n_jobs
-
-    @property
-    def n_estimators(self):
-        return len(self.regressors)
-
-    def fit(self, features, outputs):
-        """ Adds a base regressor fit on features to the ensemble
-
-        Args:
-            features: numpy array of features
-            outputs: numpy array of targets
-
-        Returns:
-
-        """
-        new_regressor = copy(self.base_regressor)
-        # each regressor needs different randomness
-        if hasattr(new_regressor, "random_state"):
-            new_regressor.random_state += len(self.regressors)
-        # loky is the process-based backend
-        with joblib.parallel_backend("loky", n_jobs=self.n_jobs):
-            new_regressor.fit(features, outputs)
-        self.regressors.append(new_regressor)
-
-    def predict(self, features):
-        """
-        Args:
-            features: 2D numpy array of features to predict on
-
-        Returns:
-            2D numpy array of predictions with N rows corresponding to N input samples.
-            Each row is the average ensemble prediction for that sample.
-        """
-        predictions = np.array(
-            [regressor.predict(features) for regressor in self.regressors]
-        )
-        return np.mean(predictions, axis=0)
-
-    def dumps(self) -> bytes:
-        batch_regressor_components = {
-            "regressors": self.regressors,
-            "base_regressor": self.base_regressor,
-            "n_jobs": self.n_jobs,
-        }
-        f = io.BytesIO()
-        joblib.dump(batch_regressor_components, f)
-        return f.getvalue()
-
-    @classmethod
-    def loads(cls, b: bytes) -> "_RegressorEnsemble":
-        f = io.BytesIO(b)
-        batch_regressor_components = joblib.load(f)
-        regressors: Sequence[sklearn.base.BaseEstimator] = batch_regressor_components[
-            "regressors"
-        ]
-        base_regressor = batch_regressor_components["base_regressor"]
-        obj = cls(
-            base_regressor=base_regressor,
-            regressors=regressors,
-            n_jobs=batch_regressor_components.get("n_jobs", 1),
-        )
-        return obj
+    def input_sensitivity(self, stacked_sample: xr.Dataset) -> InputSensitivity:
```

Comment:
Do the current offline reports include this for RFs? I haven't looked at this for a RF since before the refactor of the model. If the method isn't implemented the report just doesn't include those plots.

Reply:
They do (see `gs://vcm-ml-scratch/test-end-to-end-integration/2023-02-18/cc57ab7-a764-train-diags-prog/offline-report/sklearn_dQ1_model` for example), which is why I was confused.

Reply:
Ha never mind, it looks like the link to a report with the sensitivity figure I sent was from this PR's test (which fixed it). It does look like recent e2e tests didn't include that plot for the RF.